### PR TITLE
feat: SCRUM-82-DR-Orders-FE-reusable-Header

### PR DESCRIPTION
**Pull request by asdelay**
## Describe your changes

- Created reusable Header component 
- Created reusable ProfilePicture component
- Modified FooterNavigation component (got rid of console warning)
- Modified i18n file
- Modified fonts according to Figma design
- Added function for extracting initials from user’s first name
- Added new color
- Added new icon

## Issue ticket code (and/or) and link

- [SCRUM-82](https://codecraftersintership.atlassian.net/browse/SCRUM-82)

### **General**
Screenshot 2024-11-27 at 15 50 00
Screenshot 2024-11-27 at 15 49 36

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/components/Header/styles.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  height: calc(38px + 2 * 16px);
+  padding: 16px;
+  position: fixed;
+  top: 0;
+  left: 0;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  background-color: ${COLORS.white};
+`;
```
Comment (lyudad) [inline src/components/Header/styles.ts:17]: empty line after

File: src/components/Header/styles.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export const StyledPageName = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  white-space: nowrap;
+`;
```
Comment (lyudad) [inline src/components/Header/styles.ts:25]: empty line after. не нужно делать полотно кода, должны быть разделители визуальные

File: src/pages/MainPage/components/FooterNavigation/styles.ts [hunk 3]
```diff
@@ -59,25 +55,26 @@ export const BarMiddle = styled.div`
...
 `;
-export const SecondBarSide = styled.div<FirstBarSideProps>`
-  flex-grow: ${({ active }): number =>
-    active === RoutePage || active === OrdersPage ? 0 : 1};
+export const SecondBarSide = styled.div<{ $active?: number }>`
+  flex-grow: ${({ $active }): number =>
+    $active === RoutePage || $active === OrdersPage ? 0 : 1};
   background: ${COLORS.gray};
   transition: flex-grow ${transitionDuration} ease;
 `;
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/styles.ts:69]: empty line

File: src/pages/MainPage/components/FooterNavigation/styles.ts [hunk 4]
```diff
@@ -59,25 +55,26 @@ export const BarMiddle = styled.div`
...
-export const ThirdBarSide = styled.div<FirstBarSideProps>`
-  flex-grow: ${({ active }): number =>
-    active === MapPage || active === NotificationsPage ? 0 : 1};
+export const ThirdBarSide = styled.div<{ $active?: number }>`
+  flex-grow: ${({ $active }): number =>
+    $active === MapPage || $active === NotificationsPage ? 0 : 1};
   background: ${COLORS.gray};
   transition: flex-grow ${transitionDuration} ease;
 `;
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/styles.ts:76]: empty line

File: src/components/Header/styles.ts (hunk 1 above)
Comment (asdelay) [reply src/components/Header/styles.ts:17]: Fixed

File: src/components/Header/styles.ts (hunk 2 above)
Comment (asdelay) [reply src/components/Header/styles.ts:25]: Sorry for that. Fixed

File: src/pages/MainPage/components/FooterNavigation/styles.ts (hunk 3 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/styles.ts:69]: Fixed

File: src/pages/MainPage/components/FooterNavigation/styles.ts (hunk 4 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/styles.ts:76]: Fixed

**Pull request merged**